### PR TITLE
OnimaruShoot

### PR DESCRIPTION
**Pull request by telmiuse**
OnimaruBullet Script
TrailScreep update
OnimaruBullet Prefab
OnimaruBullet texture
UpdateSceneNavigation 

Task https://sharing.clickup.com/t/h/kd4rj3/S1FUR2F3M759TSY

### CODE REVIEW

**Comment by jcarlos0305** [comment]
Task linked: [CU-kd4rj3 Onimaru Shoot Blanace](https://app.clickup.com/t/kd4rj3)

**Review by Yonimevicio — APPROVED** [review]

**Review by jcarlos0305 — APPROVED** [review]

**Review by Jordi-Gil — DISMISSED** [review]

**Review by Jordi-Gil — APPROVED** [review]

**Review by Yonimevicio — DISMISSED** [review]

**Comment by telmiuse** [comment]
@Jordi-Gil i cant merge couse we have a problem with stretch, @lowysole is looking for the issue.

**Comment by lowysole** [comment]
> @Jordi-Gil i cant merge couse we have a problem with stretch, @lowysole is looking for the issue.

The fix will be on an Engine basis, you can continue with this PR, as the implementation seems correct

**Comment by lowysole** [comment]
> @lowysole i prefer to test it on my branch and then merge once u push the fix. Couse i need to check the script to solve the first frame of the bullet.

Gotcha

**Review by Jordi-Gil — CHANGES_REQUESTED** [review]

**Review by jcarlos0305 — DISMISSED** [review]

File: Penteract/Assets/Scripts/PlayerController.cpp [hunk 1]
```diff
@@ -285,12 +286,13 @@ void PlayerController::Shoot() {
...
-				GameplaySystems::Instantiate(onimaruTrail, onimaruGunTransform->GetGlobalPosition(), transform->GetGlobalRotation());
-				float3 frontTrail = transform->GetGlobalRotation() * float3(0.0f, 0.0f, 1.0f);
-				GameplaySystems::Instantiate(onimaruTrail, onimaruGunTransform->GetGlobalPosition(), Quat::RotateAxisAngle(frontTrail, (pi / 2)).Mul(transform->GetGlobalRotation()));
-			}			
-
+			if (onimaruBullet) {
+				GameObject* bullet = GameplaySystems::Instantiate(onimaruBullet, onimaruGunTransform->GetGlobalPosition(), Quat(0.0f,0.0f,0.0f,0.0f));
+				if (bullet) {
+					onimaruBulletcript = GET_SCRIPT(bullet, OnimaruBullet);
+					onimaruBulletcript->SetOnimaruDirection(onimaruGunTransform->GetGlobalRotation());
```
Comment (Jordi-Gil) [inline Penteract/Assets/Scripts/PlayerController.cpp:293]: Check nullity, in case bullet has not an OnimaruBullet script, this line will crash.

**Review by Jordi-Gil — APPROVED** [review]

**Review by jcarlos0305 — APPROVED** [review]

**Review by Yonimevicio — APPROVED** [review]

**Pull request merged**